docs(cloud): surface Nextflow data lineage in Run details and Data Explorer (PLAT-1255) (#1361)
seqeralabs/docs: platform-cloud/docs/data/data-explorer.md

  Select the **Path** of an object on the **View data repository** page to copy its absolute path to the clipboard. Use these object paths to specify input data locations during [pipeline launch](../launch/launchpad), add them to a [dataset](../data/datasets) for pipeline input, or when mounting data during Studio creation.

### View lineage data for objects

> When an object in Data Explorer was produced by a Nextflow run with [data lineage tracking enabled][nextflow-lineage-tutorial], the object detail view displays the object's lineage data alongside its file metadata.

Reviewer: ```suggestion
When an object in Data Explorer was produced by a Nextflow run with [data lineage tracking enabled][nextflow-lineage-tutorial], the object preview displays the object's lineage data alongside its file metadata.
```

Reviewer: There is no such thing as a "object detail view". We have "preview" only.

<!-- TODO: design to supply ./_images/data-explorer-lineage.png screenshot of the object-level lineage view. Re-add the image reference once the asset is committed: ![Object-level lineage data](./_images/data-explorer-lineage.png) -->

> Select an object on the **View data repository** page to open its detail view. When lineage data is available, the detail view displays:

Reviewer: ```suggestion
Select an object to preview. When lineage data is available, this displays:
```

> 
> | Field | Source | Description |
> |-------|--------|-------------|
> | **Lineage ID** | `LID` | Lineage ID (LID) of the object's `FileOutput` record. Select the LID to navigate to the lineage record. |
> | **Lineage Labels** | `labels` | Lineage labels assigned to the output. Each label is a clickable link to the lineage record for that label. See the Nextflow [`label` directive][nextflow-label-directive] for assignment details. |
> 

Reviewer: ```suggestion
| Field | Source | Description |
|-------|--------|-------------|
| **Lineage ID** | `LID` | Lineage ID (LID) of the object's `FileOutput` record. Click the LID to navigate to the lineage record. |
| **Lineage Labels** | `labels` | Lineage labels assigned to the output. Each label is a clickable link to the lineage record for that label. See the Nextflow [`label` directive][nextflow-label-directive] for assignment details. |
| **Produced by** | `pipeline-run` | Workflow run ID that created this object. Click the run ID to navigate to the workflow run.  |
| **Source for** | `pipeline-run` | Workflow run ID that used this file as an input. Click the run ID to navigate to the workflow run. |
—
```

If the object was not produced by a lineage-enabled run, no lineage fields are shown in the detail view.

:::tip

> Each Lineage ID and lineage label in the detail view is a navigable link. Use these links to retrace the run, task, and inputs that produced an object without leaving Seqera Platform. See [Getting started with data lineage][nextflow-lineage-tutorial] for the underlying lineage data model.

Reviewer: ```suggestion
Each Lineage ID and lineage label in the preview is a navigable link. Use these links to retrace the run, task, and inputs that produced an object without leaving Seqera Platform. See [Getting started with data lineage][nextflow-lineage-tutorial] for the underlying lineage data model.
```

> Each Lineage ID and lineage label in the detail view is a navigable link. Use these links to retrace the run, task, and inputs that produced an object — without leaving Seqera Platform. See [Getting started with data lineage][nextflow-lineage-tutorial] for the underlying lineage data model.

Reviewer 2: ```suggestion
Each Lineage ID and lineage label in the detail view is a navigable link. Use these links to retrace the run, task, and inputs that produced an object without leaving Seqera Platform. See [Getting started with data lineage][nextflow-lineage-tutorial] for the underlying lineage data model.
```

  Select the **Path** of an object on the **View data repository** page to copy its absolute path to the clipboard. Use these object paths to specify input data locations during [pipeline launch](../launch/launchpad), add them to a [dataset](../data/datasets) for pipeline input, or when mounting data during Studio creation.

### View lineage data for objects

> When an object in Data Explorer was produced by a Nextflow run with [data lineage tracking enabled][nextflow-lineage-tutorial], the object preview displays the object's lineage data alongside its file metadata.

Reviewer 3: I wouldn't point to the nextflow lineage tutorial as a way to integrate lineage in platform.
In fact platform needs to do something special about the lineage config to make sure that it will correctly capture the data, a limitation that we might lift in the future.

A better pointer would be to the nextflow lineage workspace settings that we've added and maybe the toggle in the launch form.

<!-- TODO: design to supply ./_images/data-explorer-lineage.png screenshot of the object-level lineage view. Re-add the image reference once the asset is committed: ![Object-level lineage data](./_images/data-explorer-lineage.png) -->

Select an object to preview. When lineage data is available, this displays:

> | Field | Source | Description |

Reviewer 3: Current prod fields mismatch what we document here:
Fields are:
- Lineage labels
- Produced by
- Sorce for

Currently lineage ID is not dispalyed.

|-------|--------|-------------|
| **Lineage ID** | `LID` | Lineage ID (LID) of the object's `FileOutput` record. Click the LID to navigate to the lineage record. |
| **Lineage Labels** | `labels` | Lineage labels assigned to the output. Each label is a clickable link to the lineage record for that label. See the Nextflow [`label` directive][nextflow-label-directive] for assignment details. |
| **Produced by** | `pipeline-run` | Workflow run ID that created this object. Click the run ID to navigate to the workflow run.  |
| **Source for** | `pipeline-run` | Workflow run ID that used this file as an input. Click the run ID to navigate to the workflow run. |

If the object was not produced by a lineage-enabled run, no lineage fields appear in the preview.

:::tip

> Each Lineage ID and lineage label in the preview is a navigable link. Use these links to retrace the run, task, and inputs that produced an object, without leaving Seqera Platform.

Reviewer: ```suggestion
Each lineage ID, lineage label, produced by, and source for in the preview is a navigable link. Use these links to retrace the run, task, inputs that produced an object, or outputs created by the object without leaving Seqera Platform.
```

Reviewer 3: In prod the navigable links are only produced by and source for.

labels are not navigable, and lineage id is not currently displayed

> To capture lineage data, lineage must be enabled for the run that produced the object. Enable lineage from [**Workspace settings → Lineage**][workspace-lineage-settings], the launch form toggle, or by setting `lineage.enabled = true` in your Nextflow configuration. See [Getting started with data lineage][nextflow-lineage-tutorial] for the underlying lineage data model.

Reviewer 3: ```suggestion
To capture lineage data, lineage must be enabled for the run that produced the object. Enable lineage from [**Workspace settings → Lineage**][workspace-lineage-settings] or the launch form lineage toggle. See [Getting started with data lineage][nextflow-lineage-tutorial] for the underlying lineage data model.
```
